test/e2e/scale/azdisk_scale_test.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale
+
+import (
+	"strconv"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/testsuites"
+	e2eutils "sigs.k8s.io/azuredisk-csi-driver/test/e2e/utils"
+)
+
+var _ = ginkgo.Describe("Scale testing", func() {
+
+	ginkgo.Context("[single-az]", func() {
+		scaleTests(false)
+	})
+
+})
+
+func scaleTests(isMultiZone bool) {
+	f := framework.NewDefaultFramework("azuredisk")
+
+	var (
+		cs         clientset.Interface
+		ns         *v1.Namespace
+		testDriver driver.DynamicPVTestDriver
+	)
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		testDriver = driver.InitAzureDiskDriver()
+	})
+
+	ginkgo.It("Scale test scheduling and starting multiple pods with a persistent volume requests. [Windows]", func() {
+		e2eutils.SkipIfUsingInTreeVolumePlugin()
+		e2eutils.SkipIfNotUsingCSIDriverV2()
+
+		pods := []testsuites.PodDetails{}
+		podCount := 1000
+		for itr := 0; itr < podCount; itr++ {
+			volume := testsuites.VolumeDetails{
+				FSType:    "ext3",
+				ClaimSize: "10Gi",
+				VolumeMount: testsuites.VolumeMountDetails{
+					NameGenerate:      "test-volume-" + strconv.Itoa(itr),
+					MountPathGenerate: "/mnt/test-" + strconv.Itoa(itr),
+				},
+			}
+
+			pod := testsuites.PodDetails{
+				Cmd:       e2eutils.ConvertToPowershellorCmdCommandIfNecessary("echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data"),
+				Volumes:   e2eutils.NormalizeVolumes([]testsuites.VolumeDetails{volume}, []string{}, isMultiZone),
+				IsWindows: e2eutils.IsWindowsCluster,
+			}
+
+			pods = append(pods, pod)
+
+		}
+		test := testsuites.PodSchedulingWithPVScaleTest{}
+
+		test.CSIDriver = testDriver
+		test.Pods = pods
+		test.PodCount = podCount
+		test.StorageClassParameters = map[string]string{"skuName": "StandardSSD_LRS"}
+
+		test.Run(cs, ns, schedulerName)
+	})
+
+	ginkgo.It("Scale test scheduling and rescheduling multiple pod with a persistent volume request on failover. [Windows]", func() {
+		e2eutils.SkipIfUsingInTreeVolumePlugin()
+		e2eutils.SkipIfNotUsingCSIDriverV2()
+
+		podCount := 1000
+		pods := []testsuites.PodDetails{}
+
+		for itr := 0; itr < podCount; itr++ {
```

Comment:
This loop is for number of volumes per pod. We shouldn't be iterating on number of pods here.

Reply:
This is true for most of the test cases here.

Reply:
Fixed